app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -534,6 +615,12 @@ private void setBottomAppBarButtonState() {
             bottomAppBarPreviousButton.setVisibility(View.VISIBLE);
             bottomAppBarNextButton.setVisibility(View.VISIBLE);
         }
+
+        if (hasBalance(loyaltyCard)) {
+            bottomAppBarUpdateBalanceButton.setVisibility(View.VISIBLE);
+        } else {
+            bottomAppBarUpdateBalanceButton.setVisibility(View.GONE);
+        }
```

Comment:
Just a nitpick but I'm a huge fan of ternaries for stuff like this, feels super fast to read and removes some repetition:

```suggestion
        bottomAppBarUpdateBalanceButton.setVisibility(hasBalance(loyaltyCard) ? View.VISIBLE : View.GONE);
```